src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/CSharpDiagnosticWorker.cs

```diff
@@ -186,7 +186,14 @@ private static async Task<ImmutableArray<Diagnostic>> GetDiagnosticsForDocument(
 
         public ImmutableArray<DocumentId> QueueAllDocumentsForDiagnostics()
         {
-            var documents = _workspace.CurrentSolution.Projects.SelectMany(x => x.Documents).ToImmutableArray();
+            var documents = _workspace.CurrentSolution.Projects.SelectMany(x => x.Documents);
+            QueueForDiagnosis(documents.Select(x => x.FilePath).ToImmutableArray());
+            return documents.Select(x => x.Id).ToImmutableArray();
+        }
+
+        public ImmutableArray<DocumentId> QueueDocumentsOfProjectsForDiagnostics(ImmutableArray<ProjectId> projectIds)
```

Comment:
it's probably nitpicking but I don't think you need to express again in the name what is already expressed in the argument. It could be simply an overload of the other method - `QueueAllDocumentsForDiagnostics(ImmutableArray<ProjectId> projectIds)`

Reply:
or both can be called `QueueDocumentsForDiagnostics`